sdk/messaging/azservicebus/internal/atom/models.go

```diff
@@ -54,9 +52,9 @@ type (
 		MessageCount                        *int64        `xml:"MessageCount,omitempty"`                        // MessageCount - The number of messages in the queue.
 		IsAnonymousAccessible               *bool         `xml:"IsAnonymousAccessible,omitempty"`
 		Status                              *EntityStatus `xml:"Status,omitempty"`
-		AccessedAt                          *date.Time    `xml:"AccessedAt,omitempty"`
-		CreatedAt                           *date.Time    `xml:"CreatedAt,omitempty"`
-		UpdatedAt                           *date.Time    `xml:"UpdatedAt,omitempty"`
+		AccessedAt                          string        `xml:"AccessedAt,omitempty"`
```

Comment:
Does `*time.Time` not work here?  I thought the default format is RFC3339 which is what `StringToTime()` is doing.

Reply:
So I was under the assumption that something reasonable would happen as well. Prior to my change we were actually using the date.Time type and that had a custom unmarshaller. Using just plain time.Time worked for some of the entities, but failed for others.

The only thing that consistently worked as explicitly specifying the format, which seemed easier to do this way (since it's only deserializing, not serializing).

Reply:
Do you know which ones it didn't work on?  That leads me to believe that some cases aren't in RFC3339 format which also means the `StringToTime` will silently fail.

Reply:
There is _one_ that's not parseable: `0001-01-01T00:00:00`. 

It comes back in a context where we don't even use it, so it's doubly annoying. I've modified the code to accomodate it (and return errors where appropriate).

Reply:
(also, there's no longer a silent time.Zero's)